### PR TITLE
Separate node failure maintenance from failing

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailureStatusUpdater.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.counting;
+
+/**
+ * Checks if nodes are responding and updates their status accordingly
+ *
+ * @author bratseth
+ */
+public class NodeFailureStatusUpdater extends NodeRepositoryMaintainer {
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailureStatusUpdater.java:34]: Consider a shorter name. `NodeHealthTracker`?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java [hunk 2]
```diff
@@ -385,6 +385,18 @@ private NodeAcl getNodeAcl(Node node, NodeList candidates) {
...
+    /**
+     * Returns whether the zone managed by this node repository seems to be working.
+     * If too many nodes are not responding, there is probably some zone-wide issue
+     * and we should probably refrain from making changes to it.
+     */
+    public boolean isWorking() {
+        NodeList activeNodes = list(State.active);
+        if (activeNodes.size() <= 5) return true; // Not enough data to decide
+        NodeList downNodes = activeNodes.down();
+        return ! ( (double)downNodes.size() / (double)activeNodes.size() > 0.2 );
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java:397]: Consider simplifying to `(double)downNodes.size() / (double)activeNodes.size() <= 0.2`.

**Review by mpolden — APPROVED** [review]
Great!

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java (hunk 2 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java:397]: I find "working if not more than 20% fails" is easier to get than the simpler one but that might just be me ...

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailureStatusUpdater.java (hunk 1 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailureStatusUpdater.java:34]: Good suggestion, I'll change it.

**Pull request merged**